arraycontext/impl/pytato/__init__.py

```diff
@@ -387,7 +387,19 @@ def get_target(self):
                     self.using_svm and dev.type & cl.device_type.GPU
                     and cl_char.has_coarse_grain_buffer_svm(dev))):
 
-            limit = dev.max_parameter_size
+            if dev.max_parameter_size == 4352:
```

Comment:
Add a check that there's an "Nvidia" anywhere in the platform or device name.

Reply:
I think this won't work easily since naming across pocl/Nvidia CL is a bit wonky, e.g.

```
<pyopencl.Device 'NVIDIA GeForce GTX TITAN X' on 'Portable Computing Language' at 0x55986cb63fb0>
```

but

```
<pyopencl.Device 'Tesla K40c' on 'Portable Computing Language' at 0x55cd4e26cc20>
```

(ie, no "Nvidia" in the names on the K40)

We could add 'Tesla' to the check, and hope for the best, but imho, another device having 4352 as the limit is really unlikely, so maybe it is enough.

Reply:
Ok, fair. But do add a comment about your difficulty adding a naming-based criterion.

Reply:
Added in https://github.com/inducer/arraycontext/pull/198/commits/84587db2c4f65ea7875e8d56c98f6a561fbce9ed